components/Footer.js

```diff
@@ -40,6 +63,31 @@ function Footer() {
             Create
           </NavItem>
         </Link>
+
+        <NavItem
+          isActive={false}
+          onClick={() => {
+            if (session) {
+              signOut({
+                callbackUrl: "/",
+              });
+            } else {
+              signIn();
+            }
+          }}
+        >
+          <IconContext.Provider
+            value={{
+              color: "inherit",
+              size: "2em",
+              title: "arrow icon",
+              style: { justifySelf: "center", alignSelf: "end" },
+            }}
+          >
+            {session ? <TbLogout /> : <TbLogin />}
+          </IconContext.Provider>
+          {session ? "Sign out" : "Sign In"}
+        </NavItem>
```

Comment:
although this solution is fine, you are checking on 3 major parts of the component if the use is logged in or not. You could change to a single check since every part of the comment has to change if the session state changes:
```js
{session 
  ? <NavItem>//for sigining out, sign out text and icon are displayed and the item calls signOut onClick
  :  <NavItem>//for sigining in, sign in text and icon are displayed and the item calls signIn onClick
 
}
```

Reply:
Good hint! I have additionally stored the repeating button styles in a variable.